--- target/generated-sources/org/apache/maven/model/v300/io/xpp3/MavenXpp3Writer.java	2004-09-20 18:09:09.228374400 +1000
+++ target-prev/generated-sources/org/apache/maven/model/v300/io/xpp3/MavenXpp3Writer.java	2004-09-20 18:07:53.439395200 +1000
@@ -286,6 +286,15 @@
             {
                 serializer.startTag( NAMESPACE, "type" ).text( dependency.getType() ).endTag( NAMESPACE, "type" );
             }
+            if (dependency.getProperties() != null) {
+              serializer.startTag( NAMESPACE, "properties" );
+              for (java.util.Iterator i = dependency.getProperties().keySet().iterator(); i.hasNext(); )
+              {
+                String key = ( String ) i.next();
+                serializer.startTag( NAMESPACE, key ).text( dependency.getProperty(key) ).endTag( NAMESPACE, key );
+              }
+              serializer.endTag( NAMESPACE, "properties" );
+            }
             serializer.endTag( NAMESPACE, "dependency" );
         }
         serializer.endTag( NAMESPACE, "dependencies" );
